docs/simulate_riffle_shuffle.py

```diff
@@ -0,0 +1,57 @@
+import functools
```

Comment:
Huh?
This was for you to play with the algorithm, but not actually used in code, right?

Reply:
Right, I put it in the docs as a kind of notes in case we want to adjust or re-use this later on